darts/explainability/tft_explainer.py

```diff
@@ -0,0 +1,215 @@
+from typing import Optional, Sequence, Union
+
+import matplotlib.pyplot as plt
+import pandas as pd
+
+from darts import TimeSeries, concatenate
+from darts.dataprocessing.transformers import Scaler
+from darts.datasets import IceCreamHeaterDataset
+from darts.explainability.explainability import (
+    ExplainabilityResult,
+    ForecastingModelExplainer,
+)
+from darts.models import TFTModel
+
+
+class TFTExplainer(ForecastingModelExplainer):
+
+    def __init__(
+            self,
+            model: TFTModel,
+            background_series: Optional[Union[TimeSeries, Sequence[TimeSeries]]] = None,
+            background_past_covariates: Optional[
+                Union[TimeSeries, Sequence[TimeSeries]]
+            ] = None,
+            background_future_covariates: Optional[
+                Union[TimeSeries, Sequence[TimeSeries]]
+            ] = None,
+    ):
+        """
+        Explainer class for the TFT model.
+
+        Parameters
+        ----------
+        model
+            The fitted TFT model to be explained.
+        background_series
+            The background series to be used for the TFT predict method.
+        background_past_covariates
+            The past covariates to be used for the TFT predict method.
+        background_future_covariates
+            The future covariates to be used for the TFT predict method.
+
+        """
+        super().__init__(
+            model,
+            background_series,
+            background_past_covariates,
+            background_future_covariates,
+        )
+
+        self._model = model
+        self.background_series = background_series
+        self.background_past_covariates = background_past_covariates
+        self.background_future_covariates = background_future_covariates
+        self._explain_results = None
+
+    def explain(
+            self,
+            foreground_series: Optional[Union[TimeSeries, Sequence[TimeSeries]]] = None,
+            foreground_past_covariates: Optional[
+                Union[TimeSeries, Sequence[TimeSeries]]
+            ] = None,
+            foreground_future_covariates: Optional[
+                Union[TimeSeries, Sequence[TimeSeries]]
+            ] = None,
+            horizons: Optional[Sequence[int]] = None,
+            target_components: Optional[Sequence[str]] = None,
+    ) -> ExplainabilityResult:
+        super().explain(
+            foreground_series, foreground_past_covariates, foreground_future_covariates
+        )
+        if self._explain_results is None:
```

Comment:
I don't think you should cache the explainability results, they should be recomputed when explain() is called (with potentially different series)